packages/twenty-server/src/engine/workspace-manager/dev-seeder/data/services/dev-seeder-data.service.ts

```diff
@@ -299,4 +306,402 @@ export class DevSeederDataService {
+        linkedObjectMetadataId,
+        workspaceMemberId: WORKSPACE_MEMBER_DATA_SEED_IDS.TIM,
+        companyId: null,
+        personId: null,
+        noteId: null,
+        taskId: null,
+        opportunityId: null,
+        createdAt: creationDate,
+        updatedAt: creationDate,
+        happensAt: creationDate,
+      };
+
+      // Set the appropriate target field
+      linkedActivity[`${targetInfo.targetType}Id`] = targetInfo.targetId;
+      linkedActivity[`${entityType}Id`] = entitySeed.id;
+
+      linkedActivities.push(linkedActivity);
+    }
+
+    return linkedActivities;
+  }
+
+  private async getObjectMetadataIds(workspaceId: string): Promise<{
+    noteMetadataId: string;
+    taskMetadataId: string;
+  }> {
+    const noteMetadata =
+      await this.objectMetadataService.findOneWithinWorkspace(workspaceId, {
+        where: { nameSingular: 'note' },
+      });
+
+    const taskMetadata =
+      await this.objectMetadataService.findOneWithinWorkspace(workspaceId, {
+        where: { nameSingular: 'task' },
+      });
+
+    if (!noteMetadata || !taskMetadata) {
+      throw new Error('Could not find note or task metadata');
+    }
+
+    return {
+      noteMetadataId: noteMetadata.id,
+      taskMetadataId: taskMetadata.id,
+    };
+  }
+
+  private async seedTimelineActivities({
+    entityManager,
+    schemaName,
+    workspaceId,
+  }: {
+    entityManager: WorkspaceEntityManager;
+    schemaName: string;
+    workspaceId: string;
+  }) {
+    const timelineActivities: Record<string, unknown>[] = [];
+
+    const { noteMetadataId, taskMetadataId } =
+      await this.getObjectMetadataIds(workspaceId);
+
+    let activityIndex = 0;
+
+    const entityConfigs = [
+      { type: 'company', seeds: COMPANY_DATA_SEEDS },
+      { type: 'person', seeds: PERSON_DATA_SEEDS },
+      { type: 'note', seeds: NOTE_DATA_SEEDS },
+      { type: 'task', seeds: TASK_DATA_SEEDS },
+      { type: 'opportunity', seeds: OPPORTUNITY_DATA_SEEDS },
+    ];
+
+    entityConfigs.forEach(({ type, seeds }) => {
+      seeds.forEach((seed, index) => {
+        const activity = this.createTimelineActivity(type, seed, index);
+
+        timelineActivities.push(activity);
+        activityIndex++;
+
+        // For notes and tasks, create additional linked timeline activities on target objects
+        if (type === 'note' || type === 'task') {
+          const linkedObjectMetadataId =
+            type === 'note' ? noteMetadataId : taskMetadataId;
+          const linkedActivities = this.createLinkedTimelineActivities(
+            type,
+            seed,
+            index,
+            activityIndex,
+            linkedObjectMetadataId,
+          );
+
+          timelineActivities.push(...linkedActivities);
+          activityIndex += linkedActivities.length;
+        }
+      });
+    });
+
+    if (timelineActivities.length > 0) {
+      const batchSize = 100;
+
+      for (let i = 0; i < timelineActivities.length; i += batchSize) {
```

Comment:
we rarely do for loops in the codebase unless we can't avoid it, prefer chunk() and for of (which is supporting async calls)